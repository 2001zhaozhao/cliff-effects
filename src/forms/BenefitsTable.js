// REACT COMPONENTS
import React from 'react';
import {
  Table,
  Header
} from 'semantic-ui-react';

// CUSTOM COMPONENTS
// Both the table and graph should just be added to a results page, but
// this will do for now
import ResultsGraph from '../resultsGraph';

// BENEFIT LOGIC
import { getSNAPBenefits } from '../programs/state/massachusetts/snap';
import { getHousingBenefit } from '../programs/state/massachusetts/housing';


const getSignSymbol = function ( num ) {
  if ( num > 0 ) { return '+'; }
  else if ( num < 0 ) { return '-'; }
  else { return ''; }
};  // End getSignSymbol()


const BenefitsTable = function ( props ) {

  var client        = props.client,
      currentClient = { ...client };
  currentClient.futureEarnedIncomeMonthly = currentClient.currentEarnedIncomeMonthly;

  var SNAPBenefitCurrent  = Math.round( getSNAPBenefits( currentClient ).benefitValue * 12 ),
      SNAPBenefitFuture   = Math.round( getSNAPBenefits( client ).benefitValue * 12 ),
      SNAPDiff            = SNAPBenefitFuture - SNAPBenefitCurrent,
      sec8BenefitCurrent  = Math.round( getHousingBenefit( currentClient ).benefitValue * 12 ),
      sec8BenefitFuture   = Math.round( getHousingBenefit( client ).benefitValue * 12 ),
      sec8Diff            = sec8BenefitFuture - sec8BenefitCurrent,
      totalBenefitCurrent = SNAPBenefitCurrent + sec8BenefitCurrent,
      totalBenefitFuture  = SNAPBenefitFuture + sec8BenefitFuture,
      totalDiff           = SNAPDiff + sec8Diff,
      incomeCurrent       = Math.round( client.currentEarnedIncomeMonthly * 12 ),
      incomeFuture        = Math.round( client.futureEarnedIncomeMonthly * 12 ),
      incomeDiff          = incomeFuture - incomeCurrent,
      netCurrent          = totalBenefitCurrent + incomeCurrent,
      netFuture           = totalBenefitFuture + incomeFuture,
      netDiff             = totalDiff + incomeDiff;

console.log(SNAPDiff);



const cellStyle = function(entry) {

  const   basicCellStyle    = {textAlign: 'right'}
        , columnHeaderStyle = {background: 'rgba(0, 128, 128, 1)',
                              color: 'white',
                              fontSize: '1.3em',
                              fontWeight: 900,
                              textAlign: 'center',
                              borderRadius: 'inherit',
                              letterSpacing: '0.02em'
                              }
        , rowHeaderStyle    = {fontSize: '1.2em',
                              fontWeight: 700,
                              textAlign: 'left'}
        , totalsRowStyle    = {borderTop: "2px solid rgba(0, 128, 128, 1)"};


    switch(entry) {
        case 'basicCellStyle': 
          return basicCellStyle; 
        case 'columnHeaderStyle':
          return columnHeaderStyle; 
        case 'rowHeaderStyle' :
          return  Object.assign(basicCellStyle, rowHeaderStyle);
        case 'totalsRowStyle':
          return {function(){
                    Object.assign(basicCellStyle, totalsRowStyle);
                    }
                  };
        default: 
          return basicCellStyle;
                  }
  };


const Cell = function(props) {

  return (
    <Table.Cell width={3} style={cellStyle(props.style)}> {props.value} </Table.Cell>
  );
};



const Row = function(props) {
  const numbers = props.numbers;
  const cell = numbers.map((D) =>
    <Cell key={D.index}
        value={D} />
  );

  return (
    <Table.Row>
      {cell}
    </Table.Row>
  );
}

const numbers = ['Benefit', 'Current Benefits', 'New Estimate', 'Difference'];
const numbers1 = ['SNAP', SNAPBenefitCurrent, SNAPBenefitFuture, Math.abs(SNAPDiff)];
const numbers2 = ['Section 8 Housing', sec8BenefitCurrent, sec8BenefitFuture, Math.abs(sec8Diff)];
const numbers3 = ['Total Benefits', totalBenefitCurrent, totalBenefitFuture, Math.abs(totalDiff)];
const numbers4 = ['Income', incomeCurrent, incomeFuture, Math.abs(incomeDiff)];
const numbers5 = ['Net Total', netCurrent, netFuture, Math.abs(netDiff)];




  return (
    <wrapper>


      <Table celled borderRadius='0.25em 0.25em 0 0'>
        <Table.Body >
            <Row numbers={numbers} />
            <Row numbers={numbers1} />
            <Row numbers={numbers2} />
            <Row numbers={numbers3} />
            <Row numbers={numbers4} />
            <Row numbers={numbers5} />
        </Table.Body>
      </Table>

{/*
      <Table celled>
      <Table.Body>
       <Table.Header>  Removed for formatting purposes temporarily}
          <Table.Row textAlign='center' style={columnHeaderStyle}>
            <Table.Cell width={3}>Benefit</Table.Cell>
            <Table.Cell width={3}>Current Benefits</Table.Cell>
            <Table.Cell width={3}>New Estimate</Table.Cell>
            <Table.Cell width={3}>Difference</Table.Cell>
        </Table.Row>
        </Table.Header>
      </Table.Body>
        <Table.Body>
            <Table.Row>
              <Table.Cell textAlign='left'><Header as='h4'>SNAP</Header></Table.Cell>
              <Table.Cell textAlign='right'>${SNAPBenefitCurrent} / year</Table.Cell>
              <Table.Cell textAlign='right'>${SNAPBenefitFuture} / year</Table.Cell>
              <Table.Cell textAlign='right'>{ getSignSymbol(SNAPDiff) } ${Math.abs(SNAPDiff)} / year</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell textAlign='left'><Header as='h4'>Section 8 Housing</Header></Table.Cell>
              <Table.Cell textAlign='right'>${sec8BenefitCurrent} / year</Table.Cell>
              <Table.Cell textAlign='right'>${sec8BenefitFuture} / year</Table.Cell>
              <Table.Cell textAlign='right'>{ getSignSymbol(sec8Diff) } ${Math.abs(sec8Diff)} / year</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='left'><Header as='h4'>Total Benefits</Header></Table.Cell>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='right'>${totalBenefitCurrent} / year</Table.Cell>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='right'>${totalBenefitFuture} / year</Table.Cell>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='right'>{ getSignSymbol(totalDiff) } ${Math.abs(totalDiff)} / year</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell textAlign='left'><Header as='h4'>Income</Header></Table.Cell>
              <Table.Cell textAlign='right'>${incomeCurrent} / year</Table.Cell>
              <Table.Cell textAlign='right'>${incomeFuture} / year</Table.Cell>
              <Table.Cell textAlign='right'>{ getSignSymbol(incomeDiff) } ${Math.abs(incomeDiff)} / year</Table.Cell>
            </Table.Row>
            <Table.Row style={{border: 'none'}}>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='left'><Header as='h4'>Net Total</Header></Table.Cell>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='right'>${netCurrent} / year</Table.Cell>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='right'>${netFuture} / year</Table.Cell>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='right'>{ getSignSymbol(netDiff) } ${Math.abs(netDiff)} / year</Table.Cell>
            </Table.Row>
        </Table.Body>

      </Table>
    )*/}

      <br/>
      <br/>

      <ResultsGraph {...props}/>
    </wrapper>
  );

};  // End BenefitsTable(<>)


export {
	BenefitsTable
};
