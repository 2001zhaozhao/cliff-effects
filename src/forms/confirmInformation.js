// REACT COMPONENTS
import React from 'react';
import { Header } from 'semantic-ui-react';

// PROJECT COMPONENTS
import { FormPartsContainer } from './formHelpers';

const ConfirmInformation = (props) => (
		<FormPartsContainer
			title     = 'Confirm Information'
			clarifier = ''
			left      = {{name: 'Previous', func: props.previousStep}}
			right     = {{name: 'Next', func: props.nextStep}}>
			  <div>
				<Header as='span' color='teal'>Current Benefits</Header> 
				{props.client.hasSnap ? "SNAP, ": ""}
				{props.client.hasHousing ? "Section 8, ": ""}
				{props.client.hasMassHealth ? "MassHealth": ""}<br/>
				<Header as='span' color='teal'>Household Size</Header> {props.client.householdSize}<br/>
				<Header as='span' color='teal'>Household Annual Income</Header> {props.client.annualIncome}<br/>
				<Header as='span' color='teal'>Citizenship Status</Header> {props.client.citizenshipStatus == "citizen" ? "US Citizen" : "Resident" }<br/>
				<Header as='span' color='teal'>MassHealth</Header> {props.client.hasMassHealth ? "Yes, has qualifying conditions" : "No qualifying conditions"  }
			  </div>
      </FormPartsContainer>
	  )

export { ConfirmInformation };