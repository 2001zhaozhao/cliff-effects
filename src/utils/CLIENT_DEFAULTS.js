import { cloneDeep } from 'lodash';

/**
 * Default client values.
 * MUST ALWAYS BE VALID
 */
const CLIENT_DEFAULTS = {
	current: {
    // Current programs
    hasSnap:    false,
    hasHousing: false,
    // Household
    household:  [
      {
        // 'm' for member for easier transformation
        m_age:      30,  // INT
        m_role:     'head',  // 'head', 'spouse', 'member'
        m_disabled: false
      }
    ],
    // MONEY AMOUNTS
    // Income
    earnedIncome:     0,
    TAFDC:            0,
    SSI:              0,
    SSDI:             0,
    childSupportIn:   0,
    unemployment:     0,
    workersComp:      0,
    pension:          0,
    socialSecurity:   0,
    alimony:          0,
    otherIncome:      0,
    incomeExclusions: 0,
    // Expenses
    childDirectCare:                0,
    childBeforeAndAfterSchoolCare:  0,
    childTransportation:            0,
    childOtherCare:                 0,
    earnedBecauseOfChildCare: 0,  // Negative?
    childSupportPaidOut:            0,
    adultDirectCareCosts:           0,
    adultTransportationCosts:       0,
    adultOtherCareCosts:            0,
    disabledAssistance:             0,
    earnedBecauseOfAdultCare: 0,
    disabledMedicalCosts:           0,
    otherMedicalCosts:              0,
    /** @todo When client has section 8, switch this to 'housingVoucher' */
    shelter:            'homeless',  // 'homeless', 'homeowner', 'renter', 'housingVoucher'
    contractRent:       0,
    rentShare:          0,
    rent:               0,
    mortgage:           0,
    housingInsurance:   0,
    propertyTax:        0,
    climateControl:     false,
    nonHeatElectricity: false,
    phone:              false,
    fuelAssistance:     false,  // Input is 'yes', 'no'
    otherExpenses:      0
  },
  future: {},
  // userChanged: {}  // In state instead?
};

// I do not get lodash...
CLIENT_DEFAULTS.future = cloneDeep( CLIENT_DEFAULTS.current );


export { CLIENT_DEFAULTS };
