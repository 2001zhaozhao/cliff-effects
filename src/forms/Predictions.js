import React from 'react';
import { Form, Divider, Header, Tab } from 'semantic-ui-react';
import { Bar } from 'react-chartjs-2';

// PROJECT COMPONENTS
import { FormPartsContainer, IntervalColumnHeadings, CashFlowRow } from './formHelpers';
import { BenefitsTable } from './BenefitsTable';
import {
  GraphHolder,
  GrossGraph,
  BenefitGraph
} from './ResultsGraph';

// COMPONENT HELPER FUNCTIONS
import { getTimeSetter } from '../utils/getTimeSetter';
import { getBenefitTimeFrames, getIncomeTimeFrames } from '../utils/getTimeFrames';
import { getSNAPBenefits } from '../programs/federal/snap';
import { getHousingBenefit } from '../programs/massachusetts/housing';
import {
  formatAxis,
  formatLabel,
  stackedTitle
} from '../utils/charts/chartFunctions';

// DATA
import { PROGRAM_CHART_VALUES } from '../utils/charts/PROGRAM_CHART_VALUES';


// ========================================
// COMPONENTS
// ========================================
/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const IncomeForm = function ({ future, time, setClientProperty }) {

  var type = 'income';

  /**
  * As per Project Hope input, for the first prototype we're only
  * including the ability to change earned income.
  */
  return (
    <div className='field-aligner two-column'>
      <IntervalColumnHeadings type={type}/>
      <CashFlowRow
          timeState={future}
				  type={type}
				  time={time}
				  setClientProperty={setClientProperty}
				  generic='earned'
				  labelInfo='(Weekly income = hourly wage times average number of work hours per week)'>
          Earned income
      </CashFlowRow>
    </div>
  );
};  // End IncomeForm() Component

const Chart = function({ client }) {

  var curr = client.current;

  var
    { benefitCurrent: SNAPBenefitCurrent, benefitFuture: SNAPBenefitFuture } = getBenefitTimeFrames( client, 'hasSnap', getSNAPBenefits ),
    { benefitCurrent: sec8BenefitCurrent, benefitFuture: sec8BenefitFuture } = getBenefitTimeFrames( client, 'hasHousing', getHousingBenefit ),
    { incomeCurrent, incomeFuture } = getIncomeTimeFrames( client );

  var snapData    = [ SNAPBenefitCurrent, SNAPBenefitFuture ],
      housingData = [ sec8BenefitCurrent, sec8BenefitFuture ],
      incomeData  = [ incomeCurrent, incomeFuture ];

  const SNAPColor     = PROGRAM_CHART_VALUES.snap.color,
        SNAPName      = PROGRAM_CHART_VALUES.snap.name,
        section8Color = PROGRAM_CHART_VALUES.section8.color,
        section8Name  = PROGRAM_CHART_VALUES.section8.name,
        incomeColor   = PROGRAM_CHART_VALUES.income.color,
        incomeName    = PROGRAM_CHART_VALUES.income.name;

  var datasets = [{
    label: incomeName,
    backgroundColor: incomeColor,
    data: incomeData,
    fill: "origin"
  }];

  if ( curr.hasSnap ) {
    datasets.push({
      label: SNAPName,
      backgroundColor: SNAPColor,
      data: snapData
    });
  }

  if ( curr.hasHousing ) {
    datasets.push({
      label: section8Name,
      backgroundColor: section8Color,
      data: housingData
    });
  }

  const stackedBarProps = {
    data: {
      labels: incomeData,
      datasets: datasets
    },
    options: {
      title: {
        display: true,
        text: 'Money Coming In as Income Changes'
      },
      scales: {
        yAxes: [{
          stacked: true,
          scaleLabel: {
            display: true,
            labelString: 'Total Money Coming In ($)'
          },
          ticks: {
            beginAtZero: true,
            callback: formatAxis
          }
        }],
        xAxes: [{
          stacked: true,
          scaleLabel: {
            display: true,
            labelString: 'Monthly Income ($)'
          },
          ticks: {
            callback: formatAxis
          }
        }]
      },
      tooltips: {
        callbacks: {
          title: stackedTitle,
          label: formatLabel
        }
      }
    }
  };


  return (
    <Bar {...stackedBarProps} />
  );

};  // End <Chart>

const TabbedVisualizations = ({ client }) => (
  <Tab panes={[
    { menuItem: "Compare Table", render: () => <BenefitsTable client={client} /> },
    { menuItem: "Compare Chart", render: () => <Chart client={client} /> },
    { menuItem: "Gross", render: () => <GraphHolder client={client} Graph={GrossGraph} /> },
    { menuItem: "Benefits", render: () => <GraphHolder client={client} Graph={BenefitGraph} /> }
  ]}/>
);

/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
// `props` is a cloned version of the original props. References broken.
const PredictionsStep = function ( props ) {

  const setTimeProp = getTimeSetter( 'future', props.changeClient );

  /** @todo Are these titles accurate now? */
  return (
    <Form className = 'income-form flex-item flex-column'>
      <FormPartsContainer
        title     = 'Future Household Income'
        clarifier = 'How much money would your household make in the future?'
        left      = {{name: 'Previous', func: props.previousStep}}
        right     = {{name: 'Reset', func: props.resetClient}}>
          <IncomeForm setClientProperty={setTimeProp} future={props.client.future} time={'future'} />
          <Divider className='ui section divider hidden' />
          <Header as='h3' className='ui Header align centered'>How will your income affect your future benefits?</Header>
          <TabbedVisualizations client={props.client} />
      </FormPartsContainer>
    </Form>
  );
};  // End FutureIncomeStep() Component

export { PredictionsStep };
