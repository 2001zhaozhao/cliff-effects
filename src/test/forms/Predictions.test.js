import React from 'react';
import { shallow } from 'enzyme';

import { PredictionsStep } from '../../forms/Predictions';

import { CLIENT_DEFAULTS } from '../../utils/CLIENT_DEFAULTS';
import snippets from '../../localization/en.js';

test('Prediction component renders as snapshot correctly', () => {
  const navData = {
          left:   { text: 'Previous', onClick: jest.fn() },
          middle: null,
          right:  { text: 'New Client', onClick: jest.fn() },
        },
        changeClient     = jest.fn(),
        saveForm         = jest.fn(),
        askToResetClient = jest.fn(),
        openFeedback     = jest.fn(),
        formSnippets     = snippets.visitPage.predictions;

  const wrapper = shallow(
    <PredictionsStep
      currentStep       = { 5 }
      client            = { CLIENT_DEFAULTS }
      navData           = { navData }
      changeClient      = { changeClient }
      saveForm          = { saveForm }
      askToResetClient  = { askToResetClient }
      openFeedback      = { openFeedback }
      snippets          = { formSnippets } />
  );

  expect(wrapper).toMatchSnapshot();
});
