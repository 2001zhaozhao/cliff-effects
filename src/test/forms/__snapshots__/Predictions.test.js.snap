// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prediction component renders as snapshot correctly 1`] = `
<FormPartsContainer
  clarifier={null}
  formClass="predictions"
  navData={
    Object {
      "left": Object {
        "onClick": [Function],
        "text": "Previous",
      },
      "middle": null,
      "right": Object {
        "onClick": [Function],
        "text": "New Client",
      },
    }
  }
  title={
    <span
      className="snippet"
      lang="en"
    >
      What Might Happen?
    </span>
  }
>
  <IncomeForm
    future={
      Object {
        "SSDI": 0,
        "SSI": 0,
        "TAFDC": 0,
        "adultDirectCare": 0,
        "adultOtherCare": 0,
        "adultTransportation": 0,
        "alimony": 0,
        "childBeforeAndAfterSchoolCare": 0,
        "childDirectCare": 0,
        "childOtherCare": 0,
        "childSupportIn": 0,
        "childSupportPaidOut": 0,
        "childTransportation": 0,
        "climateControl": false,
        "contractRent": 0,
        "disabledAssistance": 0,
        "disabledMedical": 0,
        "earned": 0,
        "earnedBecauseOfAdultCare": 0,
        "earnedBecauseOfChildCare": 0,
        "fuelAssistance": false,
        "hasSection8": false,
        "hasSnap": false,
        "household": Array [
          Object {
            "m_age": 30,
            "m_disabled": false,
            "m_role": "head",
          },
        ],
        "housing": "homeless",
        "housingInsurance": 0,
        "mortgage": 0,
        "nonHeatElectricity": false,
        "otherExpensesCable": 0,
        "otherExpensesCareProducts": 0,
        "otherExpensesClothes": 0,
        "otherExpensesEntertainment": 0,
        "otherExpensesFood": 0,
        "otherExpensesMedical": 0,
        "otherExpensesOther": 0,
        "otherExpensesPhone": 0,
        "otherExpensesTransport": 0,
        "otherExpensesUtilities": 0,
        "otherIncome": 0,
        "otherMedical": 0,
        "pension": 0,
        "phone": false,
        "propertyTax": 0,
        "rent": 0,
        "rentShare": 0,
        "socialSecurity": 0,
        "unemployment": 0,
        "wantsToSeeOtherExpenses": false,
        "workersComp": 0,
      }
    }
    snippets={
      Object {
        "i_afterMoney": <span
          className="snippet"
          lang="en"
      >
           / month
      </span>,
        "i_allMoneyComingIn": <span
          className="snippet"
          lang="en"
      >
          All Money Coming In as Income Changes
      </span>,
        "i_beforeMoney": <span
          className="snippet"
          lang="en"
      >
          $
      </span>,
        "i_benefitProgramsTitle": <span
          className="snippet"
          lang="en"
      >
          Individual Benefit Amounts for Household as Income Changes
      </span>,
        "i_benefitValue": <span
          className="snippet"
          lang="en"
      >
          Benefit Value ($)
      </span>,
        "i_benefitsTableTitle": <span
          className="snippet"
          lang="en"
      >
          Changes
      </span>,
        "i_buttonMonthly": <span
          className="snippet"
          lang="en"
      >
          Monthly
      </span>,
        "i_buttonWeekly": <span
          className="snippet"
          lang="en"
      >
          Weekly
      </span>,
        "i_buttonYearly": <span
          className="snippet"
          lang="en"
      >
          Yearly
      </span>,
        "i_chartsHeader": <span
          className="snippet"
          lang="en"
      >
          With the new pay, how could your benefits change?
      </span>,
        "i_columnBenefit": <span
          className="snippet"
          lang="en"
      >
          Benefit
      </span>,
        "i_columnCurrentBenefits": <span
          className="snippet"
          lang="en"
      >
          Current Benefits
      </span>,
        "i_columnDifference": <span
          className="snippet"
          lang="en"
      >
          Difference
      </span>,
        "i_columnNewEstimate": <span
          className="snippet"
          lang="en"
      >
          New Estimate
      </span>,
        "i_feedbackAsk": <span
          className="snippet"
          lang="en"
      >
          Are these numbers right? Please 
      </span>,
        "i_futureIncomeLine": <span
          className="snippet"
          lang="en"
      >
          Future Income
      </span>,
        "i_futureIncomeQuestion": <span
          className="snippet"
          lang="en"
      >
          How much money would you get paid in the future? (You can try different amounts)
      </span>,
        "i_hasIncome": <span
          className="snippet"
          lang="en"
      >
          Income
      </span>,
        "i_hasSNAP": <span
          className="snippet"
          lang="en"
      >
          SNAP
      </span>,
        "i_hasSection8": <span
          className="snippet"
          lang="en"
      >
          Section 8
      </span>,
        "i_moneyInAsIncomeChanges": <span
          className="snippet"
          lang="en"
      >
          Money Coming In as Income Changes
      </span>,
        "i_monthlyIncome": <span
          className="snippet"
          lang="en"
      >
          Monthly Income ($)
      </span>,
        "i_rowIncome": <span
          className="snippet"
          lang="en"
      >
          Income
      </span>,
        "i_rowNetTotal": <span
          className="snippet"
          lang="en"
      >
          Net Total
      </span>,
        "i_rowSNAP": <span
          className="snippet"
          lang="en"
      >
          SNAP
      </span>,
        "i_rowSection8": <span
          className="snippet"
          lang="en"
      >
          Section 8 Housing
      </span>,
        "i_rowTotalBenefits": <span
          className="snippet"
          lang="en"
      >
          Total Benefits
      </span>,
        "i_stackedAreaGraphTitle": <span
          className="snippet"
          lang="en"
      >
          Stacked Incomes
      </span>,
        "i_stackedBarGraphTitle": <span
          className="snippet"
          lang="en"
      >
          Changes Chart
      </span>,
        "i_submitFeedback": <span
          className="snippet"
          lang="en"
      >
          let us know
      </span>,
        "i_tabTitleBenefitPrograms": <span
          className="snippet"
          lang="en"
      >
          Benefit Programs
      </span>,
        "i_tabTitleChanges": <span
          className="snippet"
          lang="en"
      >
          Changes
      </span>,
        "i_tabTitleChangesChart": <span
          className="snippet"
          lang="en"
      >
          Changes Chart
      </span>,
        "i_tabTitleStackedIncomes": <span
          className="snippet"
          lang="en"
      >
          Stacked Incomes
      </span>,
        "i_title": <span
          className="snippet"
          lang="en"
      >
          What Might Happen?
      </span>,
        "i_totalMoney": <span
          className="snippet"
          lang="en"
      >
          Total Money Coming In ($)
      </span>,
        "i_warningMessage": <span
          className="snippet"
          lang="en"
      >
          This tool is in testing and these numbers might not be right. If they're not, please 
      </span>,
        "i_weeklyIncome": <span
          className="snippet"
          lang="en"
      >
          Weekly Income ($)
      </span>,
        "i_yearlyIncome": <span
          className="snippet"
          lang="en"
      >
          Yearly Income ($)
      </span>,
        "langCode": "en",
      }
    }
    time="future"
    updateClientValue={[Function]}
  />
  <Divider
    className="ui section divider hidden"
  />
  <Header
    as="h3"
    className="ui Header align centered"
  >
    <span
      className="snippet"
      key="en229"
      lang="en"
    >
      With the new pay, how could your benefits change?
    </span>
  </Header>
  <Message
    style={
      Object {
        "textAlign": "center",
      }
    }
    visible={true}
    warning={true}
  >
    <span
      className="snippet"
      key="en230"
      lang="en"
    >
      This tool is in testing and these numbers might not be right. If they're not, please 
    </span>
    <Button
      as="button"
      color="teal"
      compact={true}
      onClick={[Function]}
      size="small"
    >
      <span
        className="snippet"
        key="en232"
        lang="en"
      >
        let us know
      </span>
    </Button>
  </Message>
  <TabbedVisualizations
    client={
      Object {
        "current": Object {
          "SSDI": 0,
          "SSI": 0,
          "TAFDC": 0,
          "adultDirectCare": 0,
          "adultOtherCare": 0,
          "adultTransportation": 0,
          "alimony": 0,
          "childBeforeAndAfterSchoolCare": 0,
          "childDirectCare": 0,
          "childOtherCare": 0,
          "childSupportIn": 0,
          "childSupportPaidOut": 0,
          "childTransportation": 0,
          "climateControl": false,
          "contractRent": 0,
          "disabledAssistance": 0,
          "disabledMedical": 0,
          "earned": 0,
          "earnedBecauseOfAdultCare": 0,
          "earnedBecauseOfChildCare": 0,
          "fuelAssistance": false,
          "hasSection8": false,
          "hasSnap": false,
          "household": Array [
            Object {
              "m_age": 30,
              "m_disabled": false,
              "m_role": "head",
            },
          ],
          "housing": "homeless",
          "housingInsurance": 0,
          "mortgage": 0,
          "nonHeatElectricity": false,
          "otherExpensesCable": 0,
          "otherExpensesCareProducts": 0,
          "otherExpensesClothes": 0,
          "otherExpensesEntertainment": 0,
          "otherExpensesFood": 0,
          "otherExpensesMedical": 0,
          "otherExpensesOther": 0,
          "otherExpensesPhone": 0,
          "otherExpensesTransport": 0,
          "otherExpensesUtilities": 0,
          "otherIncome": 0,
          "otherMedical": 0,
          "pension": 0,
          "phone": false,
          "propertyTax": 0,
          "rent": 0,
          "rentShare": 0,
          "socialSecurity": 0,
          "unemployment": 0,
          "wantsToSeeOtherExpenses": false,
          "workersComp": 0,
        },
        "future": Object {
          "SSDI": 0,
          "SSI": 0,
          "TAFDC": 0,
          "adultDirectCare": 0,
          "adultOtherCare": 0,
          "adultTransportation": 0,
          "alimony": 0,
          "childBeforeAndAfterSchoolCare": 0,
          "childDirectCare": 0,
          "childOtherCare": 0,
          "childSupportIn": 0,
          "childSupportPaidOut": 0,
          "childTransportation": 0,
          "climateControl": false,
          "contractRent": 0,
          "disabledAssistance": 0,
          "disabledMedical": 0,
          "earned": 0,
          "earnedBecauseOfAdultCare": 0,
          "earnedBecauseOfChildCare": 0,
          "fuelAssistance": false,
          "hasSection8": false,
          "hasSnap": false,
          "household": Array [
            Object {
              "m_age": 30,
              "m_disabled": false,
              "m_role": "head",
            },
          ],
          "housing": "homeless",
          "housingInsurance": 0,
          "mortgage": 0,
          "nonHeatElectricity": false,
          "otherExpensesCable": 0,
          "otherExpensesCareProducts": 0,
          "otherExpensesClothes": 0,
          "otherExpensesEntertainment": 0,
          "otherExpensesFood": 0,
          "otherExpensesMedical": 0,
          "otherExpensesOther": 0,
          "otherExpensesPhone": 0,
          "otherExpensesTransport": 0,
          "otherExpensesUtilities": 0,
          "otherIncome": 0,
          "otherMedical": 0,
          "pension": 0,
          "phone": false,
          "propertyTax": 0,
          "rent": 0,
          "rentShare": 0,
          "socialSecurity": 0,
          "unemployment": 0,
          "wantsToSeeOtherExpenses": false,
          "workersComp": 0,
        },
      }
    }
    openFeedback={[Function]}
    snippets={
      Object {
        "i_afterMoney": <span
          className="snippet"
          lang="en"
      >
           / month
      </span>,
        "i_allMoneyComingIn": <span
          className="snippet"
          lang="en"
      >
          All Money Coming In as Income Changes
      </span>,
        "i_beforeMoney": <span
          className="snippet"
          lang="en"
      >
          $
      </span>,
        "i_benefitProgramsTitle": <span
          className="snippet"
          lang="en"
      >
          Individual Benefit Amounts for Household as Income Changes
      </span>,
        "i_benefitValue": <span
          className="snippet"
          lang="en"
      >
          Benefit Value ($)
      </span>,
        "i_benefitsTableTitle": <span
          className="snippet"
          lang="en"
      >
          Changes
      </span>,
        "i_buttonMonthly": <span
          className="snippet"
          lang="en"
      >
          Monthly
      </span>,
        "i_buttonWeekly": <span
          className="snippet"
          lang="en"
      >
          Weekly
      </span>,
        "i_buttonYearly": <span
          className="snippet"
          lang="en"
      >
          Yearly
      </span>,
        "i_chartsHeader": <span
          className="snippet"
          lang="en"
      >
          With the new pay, how could your benefits change?
      </span>,
        "i_columnBenefit": <span
          className="snippet"
          lang="en"
      >
          Benefit
      </span>,
        "i_columnCurrentBenefits": <span
          className="snippet"
          lang="en"
      >
          Current Benefits
      </span>,
        "i_columnDifference": <span
          className="snippet"
          lang="en"
      >
          Difference
      </span>,
        "i_columnNewEstimate": <span
          className="snippet"
          lang="en"
      >
          New Estimate
      </span>,
        "i_feedbackAsk": <span
          className="snippet"
          lang="en"
      >
          Are these numbers right? Please 
      </span>,
        "i_futureIncomeLine": <span
          className="snippet"
          lang="en"
      >
          Future Income
      </span>,
        "i_futureIncomeQuestion": <span
          className="snippet"
          lang="en"
      >
          How much money would you get paid in the future? (You can try different amounts)
      </span>,
        "i_hasIncome": <span
          className="snippet"
          lang="en"
      >
          Income
      </span>,
        "i_hasSNAP": <span
          className="snippet"
          lang="en"
      >
          SNAP
      </span>,
        "i_hasSection8": <span
          className="snippet"
          lang="en"
      >
          Section 8
      </span>,
        "i_moneyInAsIncomeChanges": <span
          className="snippet"
          lang="en"
      >
          Money Coming In as Income Changes
      </span>,
        "i_monthlyIncome": <span
          className="snippet"
          lang="en"
      >
          Monthly Income ($)
      </span>,
        "i_rowIncome": <span
          className="snippet"
          lang="en"
      >
          Income
      </span>,
        "i_rowNetTotal": <span
          className="snippet"
          lang="en"
      >
          Net Total
      </span>,
        "i_rowSNAP": <span
          className="snippet"
          lang="en"
      >
          SNAP
      </span>,
        "i_rowSection8": <span
          className="snippet"
          lang="en"
      >
          Section 8 Housing
      </span>,
        "i_rowTotalBenefits": <span
          className="snippet"
          lang="en"
      >
          Total Benefits
      </span>,
        "i_stackedAreaGraphTitle": <span
          className="snippet"
          lang="en"
      >
          Stacked Incomes
      </span>,
        "i_stackedBarGraphTitle": <span
          className="snippet"
          lang="en"
      >
          Changes Chart
      </span>,
        "i_submitFeedback": <span
          className="snippet"
          lang="en"
      >
          let us know
      </span>,
        "i_tabTitleBenefitPrograms": <span
          className="snippet"
          lang="en"
      >
          Benefit Programs
      </span>,
        "i_tabTitleChanges": <span
          className="snippet"
          lang="en"
      >
          Changes
      </span>,
        "i_tabTitleChangesChart": <span
          className="snippet"
          lang="en"
      >
          Changes Chart
      </span>,
        "i_tabTitleStackedIncomes": <span
          className="snippet"
          lang="en"
      >
          Stacked Incomes
      </span>,
        "i_title": <span
          className="snippet"
          lang="en"
      >
          What Might Happen?
      </span>,
        "i_totalMoney": <span
          className="snippet"
          lang="en"
      >
          Total Money Coming In ($)
      </span>,
        "i_warningMessage": <span
          className="snippet"
          lang="en"
      >
          This tool is in testing and these numbers might not be right. If they're not, please 
      </span>,
        "i_weeklyIncome": <span
          className="snippet"
          lang="en"
      >
          Weekly Income ($)
      </span>,
        "i_yearlyIncome": <span
          className="snippet"
          lang="en"
      >
          Yearly Income ($)
      </span>,
        "langCode": "en",
      }
    }
  />
</FormPartsContainer>
`;
